src/llama_recipes/configs/training.py

```diff
@@ -14,6 +14,7 @@ class train_config:
     batching_strategy: str="packing" #alternative: padding
     context_length: int=4096
     gradient_accumulation_steps: int=1
+    gradient_clipping: float = 1.0
```

Comment:
@gpengzhi I wonder if we should handle the default in a way to avoid any corner case. Maybe adding a bool, gradient_clipping and gradient_clipping_threshold, then guard gradient_clipping_threshold behind  gradient_clipping to make sure we don't run in to any edge case.

Reply:
An additional `bool` parameter is added to decide whether to utilize gradient clipping during training.